src/main/utility/perf_timer.rs

```diff
@@ -0,0 +1,43 @@
+use basic_timer::{DurationType, Timer};
+use std::convert::TryInto;
+use std::time::Duration;
```

Comment:
I think it's worth considering having our own timer rather than using basic_timer. The code for basic_timer is only 29 lines, and it doesn't have a license.

Reply:
Good point; done